### PR TITLE
allow platform admins to update email addresses, nameID

### PR DESCRIPTION
**Pull request by techsmyth**
The PLATFORM_ADMIN privilege is only granted to GlobalAdmins

Minor tidy up in save method of UserService

### CODE REVIEW

File: src/domain/community/user/dto/user.dto.update.platform.settings.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    description: 'Upate the URL path for the User.',
+  })
+  nameID?: string;
+
+  @Field({
+    nullable: true,
+  })
+  @IsEmail()
+  @MaxLength(MID_TEXT_LENGTH)
+  email!: string;
```
Comment (valentinyanakiev) [inline src/domain/community/user/dto/user.dto.update.platform.settings.ts]: There is disconnect between the decorator and the property, should be:
```suggestion
  email?: string;
```

File: src/domain/community/user/user.service.ts [hunk 2]
```diff
@@ -545,12 +542,17 @@ export class UserService {
...
-      await this.userRepository.save(user);
+      await this.save(user);
       await this.setUserCache(user);
     }
 
     return user;
   }
+
+  async save(user: IUser): Promise<IUser> {
+    return await this.userRepository.save(user);
```
Comment (valentinyanakiev) [inline src/domain/community/user/user.service.ts:553]: User implements IUser, but is not the same. If a DTO has more properties from `User` than the interface , these properties won't be updated in the database.

**Review by valentinyanakiev — CHANGES_REQUESTED** [review]
Two minor things to check.

File: src/domain/community/user/user.service.ts (hunk 2 above)
Comment (techsmyth) [reply src/domain/community/user/user.service.ts:553]: I think you are correct, but this is the same pattern that we have on all entities - so we do want to fix it then we need to do it structurally - not just here. So would suggest to take it separately from this PR.

**Review by valentinyanakiev — APPROVED** [review]

**Pull request merged**